### PR TITLE
Fix chat window not closing

### PR DESCRIPTION
**Pull request by Farseer2**
### Purpose
The chat window would never close when I pressed enter or submitted a message, because the event that opens it because enter was pressed occurred later. I didn't find an issue about it, though.
Also, this change makes pressing enter without a message close the chat.

### Open Questions and Pre-Merge TODOs

- [x]  This fix is tested on the same branch it is PR'ed to.
- [x]  I correctly commented my code
- [x]  My code is indented with tabs and not spaces
- [x]  This PR does not include any unnecessary .meta, .prefab or .unity (scene) changes
- [x]  This PR does not bring up any new compile errors
- [x]  This PR has been tested in editor
- [ ]  This PR has been tested in multiplayer (with 2 clients and late joiners, if applicable)

### CODE REVIEW

File: UnityProject/Assets/Scripts/UI/ControlChat.cs [hunk 1]
```diff
@@ -73,6 +73,7 @@ public class ControlChat : MonoBehaviour
 	/// Are the channel toggles on show?
 	/// </summary>
 	private bool showChannels = false;
+	private bool debounceCloseChat = false;
```
Comment (Jibbajabbafic) [inline UnityProject/Assets/Scripts/UI/ControlChat.cs]: This debounce logic is unnecessary. Behaviour seemed identical with and without it.

Comment (Jibbajabbafic) [inline UnityProject/Assets/Scripts/UI/ControlChat.cs]: This debounce var and the logic is unnecessary, the behaviour seemed the same with and without it

File: UnityProject/Assets/Scripts/UI/ControlChat.cs [hunk 2]
```diff
@@ -126,10 +129,12 @@ private void Update()
 
 		if (UIManager.IsInputFocus)
```
Comment (Jibbajabbafic) [inline UnityProject/Assets/Scripts/UI/ControlChat.cs]: You can reduce the if statement nesting a little by combining these
```suggestion
		if (UIManager.IsInputFocus && KeyboardInputManager.IsEnterPressed())
```

**Review by Jibbajabbafic — CHANGES_REQUESTED** [review]
Some minor things you could improve in the code.

Can confirm it fixes the issue where pressing enter didn't close the chat if it wasn't in focus 👍

File: UnityProject/Assets/Scripts/UI/ControlChat.cs (hunk 1 above)
Comment (Farseer2) [reply UnityProject/Assets/Scripts/UI/ControlChat.cs]: Are you sure? On my machine the chat never closes when sending a message without it... Maybe its the script load order? Or rebound keys?

Comment (Jibbajabbafic) [reply UnityProject/Assets/Scripts/UI/ControlChat.cs]: What version of unity are you using, and what OS? Mine closes just fine without it. 
@fomalsd did you experience something like this when you tested before?

Comment (Farseer2) [reply UnityProject/Assets/Scripts/UI/ControlChat.cs]: Windows 10, 2018.3.14f - The README states 2018.3

Comment (fomalsd) [reply UnityProject/Assets/Scripts/UI/ControlChat.cs]: current behaviour on develop is like that:
pressing enter with empty message never closes chat (regardless of focused state), 
pressing esc always closes chat and shows esc menu
pressing enter with non-empty message always sends it (regardless of focused state)

**Comment by Jibbajabbafic** [comment]
@Farseer2 try it now, I removed the debounce logic from the PR

**Comment by Jibbajabbafic** [comment]
Current behaviour with this PR:
- Press Enter with chat window open and no message: 
  - Close chat window
- Press Enter with chat window open and message typed out:
  - Send message
- Press Esc with chat window open:
  - Close chat window
  - Keeps message in input box if one is typed out

**Comment by Farseer2** [comment]
All of these work - but the chat doesn't close after sending a message (Which was the problem).
If it shouldn't, line 138 should be removed

**Review by Jibbajabbafic — APPROVED** [review]

**Comment by Jibbajabbafic** [comment]
I'm happy to merge this PR as is, but I can't reproduce the issue you're having:
NnQ0BjtVTK

If you're still experiencing a problem you should make an issue for it and we'll look into it more

**Comment by Jibbajabbafic** [comment]
What's your name on discord btw?

**Comment by Farseer2** [comment]
Maybe I'm using a different unity version? Something that changes script execution order?

**Pull request merged**